mlibrary/get-this
# 2022-09-08 Updates

# Overview
This pull request relates to LIT's update-a-thon!

## NPM
`npm`, `cssnano`, and `postcss` have been updated to their latest versions (as of September 8, 2022).

## `Gemfile.lock`
When running `docker-compose run --rm web bundle install`, this error was produced:

```
Your bundle is locked to webmock (3.16.0) from rubygems repository https://rubygems.org/ or installed locally, but that
version can no longer be found in that source. That means the author of webmock (3.16.0) has removed it. You'll need to
update your bundle to a version other than webmock (3.16.0) that hasn't been removed in order to install.
ERROR: 7
```
Deleting `Gemfile.lock` and rerunning the script solved the problem.

## Updates Bundler
Updates bundler to latest version

## Design System
### Spaces
The spacing variables provided by the Design System are being deprecated with the "glow up." All instances of `var(--space-...)` have been replaced by [their actual values](https://design-system.lib.umich.edu/design-tokens/), converted to `rem` measurements.

## Github Actions
* Updates ruby in tests to use `3.1` instead of pinning to a patch version
* Updates the deploy workflows to `v2` of `deploy-to-kubernetes`

## Testing
- Run the tests to make sure they pass (`docker-compose run --rm web bundle exec rspec`).
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- [View an item](http://localhost:4567/39015051081035) and make sure it all looks fine.
- Run `docker-compose run --rm web npm run build` and make sure the styles still build.
